### DISCUSSION TITLE
only half height of font showing on SSH1106 in 4w SW SPI mode

### DISCUSSION THREAD

**Discussion by strud** [olikraus/u8g2#2190] [Q&A] [answered]
I am using an SSH1106 OLED display connected to a Teensy4.1 and running in 4W SW SPI mode with this library.

When I run the basic full buffer Hello World example only the top half of the text is shown, and text on a lower line does not appear at all.
This is a VERY fast processor (600MHz), so as this is a soft SPI implementation, I'm wondering if any delays etc need to be changed?

**Comment by olikraus** [comment]
More details would be nice: Picture of the output and complete code (including the u8g2 constructor name), which renders the output of the picture.

**Comment by strud** [comment]
`U8G2_SH1106_128X64_NONAME_1_4W_SW_SPI u8g2(U8G2_R0, SCK1, MOSI1, CS2, SDA2, _RESET);`


``` C++
if (loggerConfiguration.expansionType== EXPANSION_BOARD_GUI_SPI){
      sDevice->println("initialising OLED");
      u8g2.begin();
      u8g2.clearBuffer();					// clear the internal memory
      u8g2.setFont(u8g2_font_6x12_tr);	// choose a suitable font
      u8g2.drawStr(0,60,"HiPerDaq (ultra)");	// write something to the internal memory
      //u8g2.drawStr(0,20,"fw ver 1.11");	// write something to the internal memory
      u8g2.sendBuffer();					// transfer internal memory to the display
  }
```

![IMG_3362](https://github.com/olikraus/u8g2/assets/66097047/3b49c31f-b88b-40ca-a10a-34e0278c5587)


If I use a bigger vertical offset such as 20, there is no text.

**Reply by strud** [reply]
Here is a video of the behaviour using the GraphicsTest  example without modification, using the same constructor as above code

[https://drive.google.com/file/d/1hxe19gH6z9bQrVLmVYc2FHtk2hc47Emw/view?usp=sharing](url)

The behaviour is basically the same, graphics only appearing in the top strip of the screen.

I will build up another one to make sure it isn't a dodgy  screen.

**Reply by olikraus** [reply, marked as answer]
You must use the "F" (full buffer) constructor instead of the "1" constructor.

https://github.com/olikraus/u8g2/wiki/u8g2setupcpp#constructor-name

**Reply by strud** [reply]
Fantastic, that fixed it. Many thanks

**Reply by olikraus** [reply]
I have updated the FAQ:
https://github.com/olikraus/u8g2/blob/fe36d90a0fcea86d6901ca6f86d3b102cd1f0d6a/doc/faq.txt#L147-L153

I also added a warning into the manual:
https://github.com/olikraus/u8g2/wiki/u8g2reference#sendbuffer

**Reply by strud** [reply]
Thanks for such great responsivness.

One question though, should #2 be reveresed?

Ie instead of  2. Replace "sendBuffer" with the firstPage/nextPage loop

write 2. Replace firstPage/nextPage loop with "sendBuffer"   

?

**Reply by olikraus** [reply]
oh, yes, thanks. I will fix this.

**Reply by olikraus** [reply]
hmmm it is misleading, It's either 1 or 2... let me see

**Reply by olikraus** [reply]
Hope this is better:
https://github.com/olikraus/u8g2/blob/66da29dfb947b22f8ff1da494704845c650818a0/doc/faq.txt#L147-L152

**Reply by strud** [reply]
Clearer for sure, thanks
